integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -549,6 +556,42 @@ class BenchmarkListener(list: ListBuffer[SparkPlanNode]) extends QueryExecutionL
   }
 }
 
+class BenchSparkListener(executionMetrics: ListBuffer[StageMetrics]) extends SparkListener {
+  override def onStageCompleted(stageCompleted: SparkListenerStageCompleted): Unit = {
+    val stageInfo = stageCompleted.stageInfo
+    val taskMetrics = stageInfo.taskMetrics
+
+    val stageMetrics = stageInfo.accumulables.map(acc => Try {
+      val name = acc._2.name.getOrElse("")
+      val value = acc._2.value.getOrElse(0L).asInstanceOf[Long]
+      name -> value
+    }).filter(_.isSuccess)
+        .map(_.get)
+        .filter(_._1.nonEmpty)
```

Comment:
also here why do we filter out ones without names? the api allows you to create one without a name I would say we just pass it along with id.  I guess maybe we are thinking about only ones we create will always have names?